### DISCUSSION TITLE
Country mapping using live data from the UN OCHA feed

### DISCUSSION THREAD

**Discussion by pz-max** [pypsa-meets-earth/pypsa-earth#398] [Ideas]
## Describe the feature you'd like to see

**Explorative task:**
Country naming is occasionally an issue (we have ISO standardised country letter converter - but dealing with full names is tricky). We could implement this sweet tool https://github.com/OCHA-DAP/hdx-python-country provided by the UN to deal with it in a better way.

Description of the tool (from readme):
----------------------------------
The HDX Python Country Library provides utilities to map between country and region codes and names, to convert between currencies and to match administrative level one names from different sources.

It provides country mappings including ISO 2 and ISO 3 letter codes (ISO 3166) and regions using live official data from the [UN OCHA](https://vocabulary.unocha.org/) feed with fallbacks to an internal static file if there is any problem with retrieving data from the url. (Also it is possible to force the use of the internal static files.)

It can exact match English, French, Spanish, Russian, Chinese and Arabic. There is a fuzzy matching for English look up that can handle abbreviations in country names like Dem. for Democratic and Rep. for Republic.

Mapping administration level one names from a source to a given base set is also handled including phonetic fuzzy name matching if you are running Python 3.

It also provides currency conversion to USD from local currency.

**Comment by pz-max** [comment]
Here also an incomplete list of 2 and 3 letter country codes. We intend to move to three letter coding in future.
https://github.com/wri/global-power-plant-database/blob/master/resources/country_information.csv

**Comment by davide-f** [comment]
This discussion is an open issue actually: https://github.com/pypsa-meets-earth/pypsa-earth/issues/262

country_converter relies already on several feed, I think we just need to revise that accordingly.
I think it is a mid-level task for complexity, not good for beginners, but something afterwards.
That involves changing basically all rules so it may be better to have someone with some background and pypsa-earth knowledge

Some comments may be moved there
